core/impl/src/main/java/com/blazebit/persistence/impl/AbstractCommonQueryBuilder.java

```diff
@@ -2582,17 +2582,17 @@ public BuilderType joinDefault(String path, String alias, JoinType type) {
         return (Z) joinOnEntitySubquery(base, clazz, alias, subqueryAlias, JoinType.RIGHT);
     }
 
-    private void checkJoinPreconditions(Object path, String alias, JoinType type) {
+    private void checkJoinPreconditions(Object path, String alias, JoinType type, boolean defaultJoin) {
```

Comment:
I think that `com.blazebit.persistence.impl.AbstractFullQueryBuilder#join(java.lang.String, java.lang.String, com.blazebit.persistence.JoinType, boolean, boolean)` should also use this method.